onnx/helper.py

```diff
@@ -1029,3 +1026,48 @@ def make_training_info(algorithm: GraphProto, algorithm_bindings: AssignmentBind
             binding.value = v
 
     return training_info
+
+
+def tensor_dtype_to_np_type(tensor_dtype: int) -> np.dtype:
```

Comment:
I think it will also help to document all of these, given the subtle differences (eg., like storage_tensor_type vs storage_numpy_type, etc.). Given that the functions are inter-related, we could put it into some doc file perhaps,
just to clarify the different types, and when/where they are used or should be used.

Reply:
We could also try to reduce the number of helper functions down below, in case it helps. Not sure if all of these are used to a significant degree?

Reply:
Good point. Adding more docs for these functions should be very useful. Just added API comments in the functions and two commonly-used functions (`tensor_dtype_to_np_type` and `tensor_dtype_to_field`) in PythonAPIOverview.md. Please review it. Thanks!

> We could also try to reduce the number of helper functions down below, in case it helps. Not sure if all of these are used to a significant degree?

I think perhaps functions like `sequence_type_to_field`, `optional_type_to_field`, `tensor_dtype_to_storage_tensor_type` can be private functions, because ONNX code can directly use outer functions like `get_attr_from_sequence_elem_type`, `get_attr_from_optional_elem_type`, `tensor_dtype_to_storage_numpy_type` correspondingly. I am not very sure whether those functions are helpful to external users though. @gramalingam WDYT?